docs/best_practice/crate_ami.txt

```diff
@@ -0,0 +1,114 @@
+``AMI Name``. AMIs are listed in JSON-format and sorted by its creation date.
+The following command lists every avaialble Crate AMI on the AWS platform.
+
+.. code-block:: bash
+
+    $ aws ec2 describe-images --filters "Name=name,Values=Crate-AWS-AMI-*"
+
+
+Launch the AMI
+==============
+
+A new EC2 instance can be started using the ``run-instances`` command. Use the
+available parameters of the command to configure its start-up behavior and
+hardware infrastructure.
+
+.. code-block:: bash
+
+    $ aws ec2 run-instances --image-id <AMI-ID> --count <NR-OF-INSTANCES> --instance-type <INSTANCE-TYPE> --user-data <USER-DATA>
+
+
+If you want to enable EC2 discovery on all instances to
+**support unicast host discovery** in order that every instance can join the
+cluster it is required to configure the AWS Credentials on startup of your
+instance. In this case the ``USER-DATA`` parameter has to be set with
+the following data.
+
+.. code-block:: bash
+
+    #!/bin/bash
+    echo "
+    AWS_ACCESS_KEY_ID=###your_access_key_id###
+    AWS_SECRET_ACCESS_KEY='###your_secret_access_key###'
+    " >> /etc/sysconfig/crate
+
+This configuration script will be launched on startup of each instance.
+The following example shows how you can run a single instance of type
+``m3.medium`` and pass the AWS credentials as file in the ``user-data`` field.
+
+.. code-block:: bash
+
+    aws ec2 run-instances --image-id ami-544c1923 --count 1 --instance-type m3.medium --user-data file://my_aws_credentials
+
+
+.. note::
+    In case that you want to filter machines by instance tags or security groups
+    add this configuration also to the ``USER-DATA`` field parameter
+    (see :ref:`filter-by-tags`).
+
+
+Instance Types
+==============
+
+The instance type specifies the combination of CPU, memory, storage and
+networking capacity. To receive a better performance for running queries
+select an instance type which gives you the possibility to attach ephemeral
+storage. On newer instance types on AWS this storage is covered by
+`Solid-State-Drives`_ (short *SSD*). By choosing one of those instance types
+Crate will automatically mount and store its data on those devices if they are
+attached to the instance as a block device mapping (see also :ref:`attached_devices`).
+Instance Types with additional instance store volumes (SSD or HDD) are
+currently all instances of type ``m3``, ``g2``, ``r3``, ``d2`` and ``i2``.
+
+.. _attached_devices:
+
+Attached Devices
+================
+
+To add a block device mapping before launching an instance it is possible to
+use the ``block-device-mappings`` parameter on the ``run-instances`` command.
+In this case ``ephemeral1`` will be added as an instance store volume with the
+device name ``/dev/sdc``.
+
+.. code-block:: bash
+
+    --block-device-mappings "[{\"DeviceName\": \"/dev/sdc\",\"VirtualName\":\"ephemeral1\"}]"
+
+.. note::
+    Note that the data stored in any kind of an instance storage type is not
```

Comment:
data stored on ephemeral disks is nor permanent and only persists during the lifetime ...